### PR TITLE
new package: py-cmake

### PR DESCRIPTION
**Pull request by Sinan81**
Related to #17515

### CODE REVIEW

**Comment by adamjstewart** [comment]
Is this still a WIP?

**Comment by Sinan81** [comment]
`py-cmake` seems to be duplicating stuff provided by cmake package.  I am not sure that's acceptable. I am wondering setup.py or other source files  can be hacked to prevent that.

**Comment by adamjstewart** [comment]
> py-cmake seems to be duplicating stuff provided by cmake package.

What do you mean? It looks like a simple wrapper around an external CMake command to me.

**Comment by Sinan81** [comment]
> > py-cmake seems to be duplicating stuff provided by cmake package.
> 
> What do you mean? It looks like a simple wrapper around an external CMake command to me.

For one, it provides cmake command under `py-cmake-prefix/bin`

**Comment by adamjstewart** [comment]
I think that's fine as long as it has the same CLI as the actual cmake command.

File: var/spack/repos/builtin/packages/py-cmake/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+from spack import *
+
+
+class PyCmake(PythonPackage):
+    """CMake is an open-source, cross-platform family of tools designed to
+    build, test and package software
+    """
+
+    homepage = "https://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmake/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#homepage

File: var/spack/repos/builtin/packages/py-cmake/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    url      = "https://files.pythonhosted.org/packages/bf/b6/bc83a846ac9ffbcb7f9490af135c42002e12f3adc8253c9e55c07de49cf3/cmake-3.18.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cmake/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#url

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-cmake/package.py (hunk 1 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-cmake/package.py]: Oops.

File: var/spack/repos/builtin/packages/py-cmake/package.py (hunk 2 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-cmake/package.py]: Now python hosted links are automatically converted to pypi links right? (or let me just read the docs)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**